dll/win32/kernel32/client/session.c

```diff
@@ -13,31 +16,141 @@
 #define NDEBUG
 #include <debug.h>
 
+/* GLOBALS ********************************************************************/
+
+PBASE_STATIC_SERVER_DATA BaseStaticServerData;
+
+#define LMEM_FIXED 0x0
+
+#define ERROR_INVALID_PARAMETER 0x57
+#define ERROR_NOT_ENOUGH_MEMORY 0x8
+
 /* FUNCTIONS ******************************************************************/
 
 /*
- * @unimplemented
+ * @implemented
  */
 DWORD
 WINAPI
 DosPathToSessionPathW(IN DWORD SessionID,
                       IN LPWSTR InPath,
                       OUT LPWSTR *OutPath)
 {
-    UNIMPLEMENTED;
+    DWORD LastError = 0;
+    if (InPath == NULL)
+    {
+        LastError = ERROR_INVALID_PARAMETER;
+    }
+    else
+    {
+        size_t InPathLength = wcslen(InPath);
+        size_t OutPathLength = 0;
+        LPWSTR SessionPath = 0;
+        if (BaseStaticServerData->LUIDDeviceMapsEnabled == FALSE) {
+            if (SessionID == 0)
+            {
+                OutPathLength += wcslen(L"GLOBALROOT\\DosDevices\\");
+                OutPathLength += InPathLength;
+            }
+            else
+            {
+                OutPathLength += wcslen(L"GLOBALROOT\\Sessions\\");
+                OutPathLength += 10; // Length of SessionID(char)
+                OutPathLength += wcslen(L"\\DosDevices\\");
+                OutPathLength += InPathLength;
+            }
+        }
+        else
+        {
+            OutPathLength += InPathLength;
+        }
+        OutPathLength += 1; // \0 termination
+        SessionPath = (LPWSTR)LocalAlloc(LMEM_FIXED, OutPathLength * sizeof(WCHAR));
+        if (SessionPath != 0)
```

Comment:
`SessionPath` is a pointer, compare it with `NULL` instead of `0`. Or just `if (SessionPath)`.